packages/syft/src/syft/service/action/action_object.py

```
@@ -474,6 +575,40 @@ def make_id(cls, v: Optional[UID]) -> UID:
         """Generate or reuse an UID"""
         return Action.make_id(v)
 
+    class Config:
+        validate_assignment = True
+
+    @pydantic.root_validator()
+    def __check_action_data(cls, values: dict) -> dict:
+        v = values.get("syft_action_data_cache")
+        if values.get("syft_action_data_type", None) is None:
+            values["syft_action_data_type"] = type(v)
+        if not isinstance(v, ActionDataEmpty):
+            if inspect.isclass(v):
+                values["syft_action_data_repr_"] = repr_cls(v)
+            else:
+                values["syft_action_data_repr_"] = (
+                    v._repr_markdown_()
+                    if hasattr(v, "_repr_markdown_")
+                    else v.__repr__()
+                )
+            values["syft_action_data_str_"] = str(v)
+            values["syft_has_bool_attr"] = hasattr(v, "__bool__")
+        return values
+
+    def _save_to_blob_store(self) -> Optional[SyftError]:
+        data = self.syft_action_data
+        if isinstance(data, SyftError):
+            return data
+        if isinstance(data, ActionDataEmpty):
+            print(f"cannot store empty object {self.id}")
+            return data
```

We should return an error here as well, so we don't have to check everytime if the result is an error or an ActionDataEmpty